### PR TITLE
Vendor virtcontainers 1.0.6 and adapt the stale asset algorithm

### LINKED ISSUE #804: Pull virtcontainers 1.0.6
We need the SRIOV and multi-os changes for 3.0.7.

### PR DESCRIPTION
**Pull request by sameo**
We pull the new SRIOV and multi OS support.
    
Shortlog:
    
6403e0c 1.0.6 release
f08b6c1 hypervisor: Provide custom asset getter
ac2a937 asset: Extend asset support to firmware and hypervisor
fc5ae53 hypervisor: Add a private custom assets map
5504939 oci: Add asset annotations to the pod config
de377a9 pod: Select kernel and image assets from annotations
1d1c5f4 pkg: Add annotations package
ec2f2c3 network: Rename Endpoint interface methods.
6c31530 networking : Pass MTU information for physical network devices.
60b8c34 network: Skip setting MTU for physical interfaces.
ad92928 tests: Add test to check if interface is physical.
7ec762a tests: Add tests to check for Network Endpoint Types
96571ee tests: Modify tests to test for virtual endpoints.
dc715cf network: Add methods to attach and detach network interfaces.
bad2911 network: Handle marshalling/unmarshalling of Endpoint interface array
3355932 sriov: Bind and unbind Virtual function based interfaces
7f4a3fb .ci: Fix coveralls reports
17b7761 .ci: Remove travis CI
5e428cc container: Quieten logging
3efc9ac sriov: Add support for physical interfaces.
6a07eef network: Add function to check if interface is a physical device
5c53bbe ethtool: Add ethtool package
d2acfa1 1.0.5 release
8be82a3 ci: restore the use of chronic
b366203 .ci: Handle cases where backward compatibility is broken

Fixes #804

### CODE REVIEW

**Comment by sboeuf** [comment]
LGTM

File: list.go [hunk 1]
```diff
@@ -151,12 +157,24 @@ To list containers created using a non-default value for "--root":
 func getStaleAssets(old, new hypervisorDetails) []string {
 	var stale []string
 
-	if old.KernelPath != new.KernelPath {
-		stale = append(stale, "kernel")
+	if old.KernelAsset.Path != new.KernelAsset.Path {
+		if old.KernelAsset.Custom {
+			if _, err := os.Stat(old.KernelAsset.Path); os.IsNotExist(err) {
```
Comment (jodh-intel) [inline list.go]: Could you add a comment explaining this logic?

File: list.go [hunk 2]
```diff
@@ -151,12 +157,24 @@ To list containers created using a non-default value for "--root":
...
+			stale = append(stale, "kernel")
+		}
 	}
 
-	if old.ImagePath != new.ImagePath {
-		stale = append(stale, "image")
+	if old.ImageAsset.Path != new.ImageAsset.Path {
+		if old.ImageAsset.Custom {
+			if _, err := os.Stat(old.ImageAsset.Path); os.IsNotExist(err) {
+				stale = append(stale, "kernel")
```
Comment (jodh-intel) [inline list.go]: I think this should be `"image"`?

File: list.go [hunk 3]
```diff
@@ -208,13 +226,26 @@ func (f *formatTabular) Write(state []fullContainerState, showAll bool, file *os
...
+
+			if !current.ImageAsset.Custom {
+				all += fmt.Sprintf("\t%s", latest.ImageAsset.Path)
+			} else {
+				all += fmt.Sprintf("\t%s", current.ImageAsset.Path)
+			}
+
+			all += fmt.Sprintf("\t%s\n", stale)
+
+			fmt.Fprintf(w, all)
```
Comment (jodh-intel) [inline list.go]: This could use `fmt.Fprint()` as there isn't a format string.

**Comment by sameo** [comment]
@jodh-intel All comments addressed, thanks for the review.

**Pull request merged**